### ISSUE TITLE
`no-fallthrough` warns when infinite loop with no break precludes fallthrough

### ISSUE THREAD

**Issue by peetklecha** [denoland/deno_lint#1156]
### Lint Name
no-fallthrough

### Code Snippet

```ts
function* sequence(type: "even" | "odd" | "prime") {
  let num = 0;
  switch (type) {
    case "even":
      while (true) {
        if (isEven(num)) yield num;
        num += 1;
      }
      // break;
    case "odd":
      ...
  }
}
```

### Expected Result

There should be no warning.

### Actual Result

```
Fallthrough is not allowed
Add `break` or comment `/* falls through */` to your case statementdeno-lint(no-fallthrough)
```

Uncommenting the `break` causes that to receive an unreachable code warning, which shows that the linter already identifies the loop as never breaking.

### Additional Info

Same basic issue as #1042 but this should be even more straightforward to fix.

### Version

deno 1.33.2 (release, x86_64-apple-darwin)
v8 11.4.183.1
typescript 5.0.3

**Comment by magurotuna** [comment]
I think the core of the issue can be seen in the following snippet:

```ts
switch (someValue) {
  case 0:
    while (true) {}
  default:
    console.log(42);
}
```

The control flow analyzer does know `while (true) {}` is an infinite loop. However, this information is not utilized well by `no-fallthrough` rule - maybe the straightforward way to solve this would be that we let `no-fallthrough` skip the fallthrough check if the final point of a `case` (i.e. in the above example, the point right after `while (true) {}` but before going into the next branch, `default:`) is unreachable. This may require some fix in control flow analyzer.

**Comment by bartlomieju** [comment]
Fixed in https://github.com/denoland/deno_lint/pull/1497
